packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -341,8 +347,60 @@ export async function* OpenAIChatModel(
     promptTokenLimit
   );
 
-  const messages: OpenAIClient.Chat.CreateChatCompletionRequestMessage[] = await Promise.all(
-    conversationMessages.map(async (message) => {
+  // OpenAI requires that tool calls with IDs be immediately followed by tool messages with the corresponding IDs.
```

Comment:
Nit: a general comment that this function is now 350 lines long :)

Reply:
yeah, hoisting this ID management to a helper would be a nice simplification.